metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/KafkaDestinationHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.metron.profiler.bolt;
+
+import org.apache.metron.common.utils.JSONUtils;
+import org.apache.metron.profiler.ProfileMeasurement;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Values;
+import org.json.simple.JSONObject;
+
+import java.io.Serializable;
+
+/**
+ * Handles emitting a ProfileMeasurement to the stream which writes
+ * profile measurements to Kafka.
+ */
+public class KafkaDestinationHandler implements DestinationHandler, Serializable {
+
+  /**
+   * The stream identifier used for this destination;
+   */
+  private String streamId = "kafka";
+
+  @Override
+  public void declareOutputFields(OutputFieldsDeclarer declarer) {
+    // the kafka writer expects a field named 'message'
+    declarer.declareStream(getStreamId(), new Fields("message"));
+  }
+
+  @Override
+  public void emit(ProfileMeasurement measurement, OutputCollector collector) {
+
+    try {
+      JSONObject message = new JSONObject();
+      message.put("profile", measurement.getDefinition().getProfile());
+      message.put("entity", measurement.getEntity());
+      message.put("period", measurement.getPeriod().getPeriod());
+      message.put("periodStartTime", measurement.getPeriod().getStartTimeMillis());
+
+      // TODO How to serialize an object (like a StatisticsProvider) in a form that can be used on the other side? (Threat Triage)
+      // TODO How to embed binary in JSON?
+      message.put("value", measurement.getValue());
+
```

Comment:
Pushing the data to Kafka is *not* meant as a replacement for HBase.  You still need to persist the Profile data in HBase, just like before.  The data is not pushed to Kafka to be used for retrieval.  The data is sent to Kafka to give us a hook into the triage process.  

Let me try to explain better.  To triage something, I need to know two things...
1. What is the current value? 
2. What is an abnormal value?

#### What is the current value? 
The Profile data in Kafka helps answer the first question. What is the current value? Prior to this PR, there is no way for the Triage process to 'see' a summarized value.  I tried to explain this in the PR description (probably poorly) as follows.

> The value being interrogated here, the number of inbound flows, is not a static value contained within any single telemetry message. This value is calculated across multiple messages by the Profiler. The current Threat Triage process cannot be used to interrogate values calculated by the Profiler.

#### What is an abnormal value?
The Profile data in HBase helps answer this second question.  What is an abnormal value?  The triage rules would still use the Profiler Client and retrieve the data from HBase that defines what normal is.  I think you are referring to this in your response.

Let me know if this provides any clarity or if I am misunderstanding the question.